File: 프로그래머스/2/42586. 기능개발/기능개발.py
import math

def solution(progresses, speeds):
    answer = []
    i = 0
    while i < len(progresses):
        s = speeds[i]
        p = progresses[i]
        
        
        cnt = 0
        t = math.ceil((100 - p) / s)
        while i < len(progresses) and progresses[i] + speeds[i] * t >= 100:
            cnt += 1
            i += 1
            
        answer.append(cnt)
                
    return answer
        
        

File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    const result = []
    let idx = 0;
    let days = 0;
    while(idx < progresses.length){
        days += Math.ceil((100 - (progresses[idx] + speeds[idx] * days)) / speeds[idx]);
        let cnt = 0
        while(idx < progresses.length && progresses[idx] + speeds[idx] * days >= 100){
            cnt += 1;
            idx += 1;
        }
        result.push(cnt);
    }
    
    return result;
}